streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/PublishSubjectImpl.kt

```diff
@@ -89,11 +89,12 @@ open class PublishSubjectImpl<T> : PublishSubject<T> {
             }
     }
 
-    protected fun cleanupValues() {
+    fun cleanupValues() {
         if (hasSubscriptions) throw IllegalStateException("Cannot clean values when publisher has subscribers")
         serialQueue.dispatch {
             atomicValue.setOrThrow(atomicValue.value, null)
             atomicError.setOrThrow(atomicError.value, null)
+            isCompleted.setOrThrow(isCompleted.value, false)
```

Comment:
This has to be removed too, you should not clean values if the publisher has completed as the value should not change. We probably should assert that a publisher has not completed when cleaning values